src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -46,77 +47,97 @@ private enum GlobalNotificationState
-                catch (ConnectionLostException ex)
+                catch (Exception e)
                 {
-                    RemoteHostCrashInfoBar.ShowInfoBar(workspace, ex);
-
-                    Shutdown(ex);
-
-                    // dont crash VS because OOP is failed to start. we will show info bar telling users to restart
-                    // but never physically crash VS.
-                    return null;
-                }
-                catch (SoftCrashException ex)
-                {
-                    Shutdown(ex);
+                    // translate to our own cancellation if it is raised.
+                    cancellationToken.ThrowIfCancellationRequested();
 
-                    // at this point, we should have shown info bar (RemoteHostCrashInfoBar.ShowInfoBar) to users
-                    // returning null here will disable OOP for this VS session. 
-                    // * Note * this is not trying to recover the exception. but giving users to time
-                    // to clean up before restart VS
-                    return null;
+                    throw SoftCrash(workspace, hubClient, e, "Failed to connect to ServiceHub", cancellationToken);
                 }
-                catch (Exception ex)
+                finally
                 {
-                    Shutdown(ex);
-                    throw;
+                    if (!success)
+                    {
+                        client.Dispose();
+                    }
                 }
 
-                void Shutdown(Exception ex)
-                {
-                    // make sure we shutdown client if initializing client has failed.
-                    client?.Shutdown();
+                client.Started();
+                return client;
+            }
+        }
 
-                    // translate to our own cancellation if it is raised.
-                    cancellationToken.ThrowIfCancellationRequested();
+        public static async Task<Stream> RequestServiceAsync(
+            Workspace workspace,
+            HubClient client,
+            string serviceName,
+            HostGroup hostGroup,
+            CancellationToken cancellationToken)
+        {
+            Exception unexpectedException;
 
-                    // otherwise, report watson
-                    ex.ReportServiceHubNFW("ServiceHub creation failed");
-                }
+            var descriptor = new ServiceDescriptor(serviceName) { HostGroup = hostGroup };
+            try
+            {
+                return await client.RequestServiceAsync(descriptor, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception e) when (e is ConnectionLostException || e is RemoteInvocationException re && re.InnerException is InvalidOperationException)
+            {
+                // ServiceHub may throw these exceptions if it is called after VS started to shut down.
+                // Cancel the operation and do not report an error in these cases.
```

Comment:
i'm not certain i can reconcile this comment witht eh code below.  It doesn't look like we're canceling anything.  We're just doing cancellationToken.ThrowIfCancellationRequested.  But how do you know cancellation was requested?

Should we just throw OperationCanceledException here?  Or at least add a check that hte cancellationToken is in the 'canceled' state?